### PR TITLE
Nestjs message dispatcher

### PR DESCRIPTION
**Pull request by JERILJOSE96**

### CODE REVIEW

File: test/message-dispatcher.spec.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    }).compile();
+    app = moduleFixture.createNestApplication();
+    await app.init();
+
+    transport = app.get<AsyncLoggerProvider>(AsyncLoggerProvider);
+  });
+
+  describe('interceptor', () => {
+    it('should create and send message', async () => {
+      jest.spyOn(transport, 'log');
```
Comment (espresse) [inline test/message-dispatcher.spec.ts:62]: @JERILJOSE96  it should be in the `beforeEach` block, also it should be cleared after each test

File: test/message-dispatcher.spec.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      const id = '123-test';
```
Comment (espresse) [inline test/message-dispatcher.spec.ts]: @JERILJOSE96 keep it outside of the test case (in the respective `describe` block), we should make our tests cases as lean as possible, focused on the things you're testing - it helps reading them.

File: test/message-dispatcher.spec.ts (hunk 1 above)
Comment (JERILJOSE96) [reply test/message-dispatcher.spec.ts:62]: done

File: test/message-dispatcher.spec.ts (hunk 2 above)
Comment (JERILJOSE96) [reply test/message-dispatcher.spec.ts]: moved

**Review by espresse — CHANGES_REQUESTED** [review]
@JERILJOSE96  it looks very good by now. I think the only piece missing is some examples how to use it in case somebody else would like to use this library. Could you add some examples to README? See our other repos in github for inspiration.

**Review by espresse — APPROVED** [review]
@JERILJOSE96 LGTM :+1:

**Pull request merged**